PollBuddy-Server/backend/routes/polls.js

```diff
@@ -90,84 +90,63 @@ router.post("/:id/edit", async (req, res) => {
   return res.status(200).send(createResponse());
 });
 
-router.post("/:id/submit", function (req, res) {
-  const jsonContent = req.body;
-  const pollId = new mongoConnection.getMongo().ObjectID(req.params.id);
-  let data = {}; // Stores data being submitted to DB
-  let insert = {}; // Stores insertion location
-
-  // Check that pollId was specified and is valid
-  if (pollId !== undefined) {
-    mongoConnection.getDB().collection("polls").find({"_id": pollId}).toArray(function (err, result) {
-      if (err) {
-        return res.sendStatus(500);
-      }
-      if (result.length === 0) {
-        return res.status(500).send({"Result": "Error", "Error": "Cannot find poll"});
-      }
-    });
-  }
-
-  // Check that answers were supplied in the correct format
-  if (!jsonContent.Answers) {
-    return res.status(500).send({"Result": "Error", "Error": "Answers not specified"});
-  }
-  if (!Array.isArray(jsonContent.Answers)) {
-    return res.status(500).send({"Result": "Error", "Error": "Answers is not an array"});
-  }
-  if (jsonContent.Answers.empty) {
-    return res.status(500).send({"Result": "Error", "Error": "Answers is empty"});
-  }
-
-  // Add timestamp to answers
-  data.Answers = jsonContent.Answers;
-  data.Timestamp = Date.now();
-
-  // Check if the user is logged in or anonymous
-  if (req.session.UserID) {
-    // User is logged in, save with their ID
-    insert["$and"] = [{"PollID": pollId}, {"UserID": jsonContent.UserID}];
-  } else {
-    insert["PollID"] = pollId;
+/**
+ * Submit/re-submit poll answer.
+ * Depending on the "Pubic" attribute of the poll it might requires sign-in.
+ * @typedef {Object} Answers
+ * @property {string} QuestionID - ID of the question.
+ * @property {string} Answer - Answer response, could be null or empty string.
+ * @postdata {Answers[]} payload
+ * @throws 400 - Invalid ID.
+ * @throws 403 - Sign-In required.
+ * @throws 500 - An error occurred while reading the database.
+ * @throws 500 - An error occurred while writing to the database.
```

Comment:
I know we don't have a clear defined format for this, but could you combine the two database messages into `An error occurred while communicating with the database.`? It's not really relevant to report to the frontend which database action failed, just that a database issue occurred and therefore the task failed. That would simplify using the endpoint too a bit. 

I would like to make that a standard format though. Thoughts from anyone?